gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ProjectInfo.java

```diff
@@ -0,0 +1,220 @@
+ * the License.
+ */
+
+package com.google.gcloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A Google Cloud Resource Manager project metadata object.
+ */
+public class ProjectInfo implements Serializable {
+
+  private static final long serialVersionUID = 9148970963697734236L;
+  private final String name;
+  private final String id;
+  private final Map<String, String> labels;
+  private final Long number;
+  private final State state;
+  private final Long createTimeMillis;
+  private final ResourceId parent;
+
+  public enum State {
+    LIFECYCLE_STATE_UNSPECIFIED,
+    ACTIVE,
+    DELETE_REQUESTED,
+    DELETE_IN_PROGRESS;
+  }
+
+  public static class Builder {
+    private String name;
+    private String id;
+    private Map<String, String> labels;
+    private Long number;
+    private State state;
+    private Long createTimeMillis;
+    private ResourceId parent;
+
+    Builder() {
+      labels = new HashMap<String, String>();
+    }
+
+    public Builder name(String name) {
+      this.name = name;
+      return this;
+    }
+
+    public Builder id(String id) {
+      this.id = id;
+      return this;
+    }
+
+    public Builder addLabel(String key, String value) {
+      this.labels.put(key, value);
+      return this;
+    }
+
+    public Builder removeLabel(String key) {
+      this.labels.remove(key);
+      return this;
+    }
+
+    public Builder clearLabels() {
+      this.labels.clear();
+      return this;
+    }
+
+    public Builder labels(Map<String, String> labels) {
+      this.labels = Maps.newHashMap(checkNotNull(labels));
+      return this;
+    }
+
+    Builder number(Long number) {
+      this.number = number;
+      return this;
+    }
+
+    Builder state(State state) {
+      this.state = state;
+      return this;
+    }
+
+    Builder createTimeMillis(Long createTimeMillis) {
+      this.createTimeMillis = createTimeMillis;
+      return this;
+    }
+
+    public Builder parent(ResourceId parent) {
+      this.parent = parent;
+      return this;
+    }
+
+    public ProjectInfo build() {
+      return new ProjectInfo(this);
+    }
+  }
+
+  ProjectInfo(Builder builder) {
+    this.name = builder.name;
+    this.id = checkNotNull(builder.id);
+    this.labels = ImmutableMap.copyOf(builder.labels);
+    this.number = builder.number;
+    this.state = builder.state;
+    this.createTimeMillis = builder.createTimeMillis;
+    this.parent = builder.parent;
+  }
+
+  public String id() {
+    return id;
+  }
+
+  public String name() {
+    return name;
+  }
+
+  public Long number() {
+    return number;
+  }
+
+  public Map<String, String> labels() {
```

Comment:
Here mention that return map is immutable (use generic term, not reference to guava).